### PR TITLE
Provide simple way to generate a range of numbers

### PR DESCRIPTION
**Pull request by zpiman**
When a second parameter is added to function `number` make the first parameter `options.min` and the second `options.max`.
This makes it easy to generate a range of values: 

``` javascript
faker.random.number(10,50) // -> generates numbers 10 to 50...
```

### CODE REVIEW

**Comment by brianmcmillan** [comment]
I believe this functionality already exists. From the wiki:
Basic random data

Whole Number
faker.random.number(min,max)
Random number between 0 and "range".
faker.random.number(100); = 92
faker.random.number(5,10); = 9

Decimal number
faker.finance.amount(min,max,decimal places)
Random number between "min" and "max" including decimals to X digits. faker.finance.amount(9000,10000,4); = 9948.8363

**Comment by zpiman** [comment]
Thats what I thought as well @brianmcmillan. But it turns out it is not implemented that way. I was confused in my project why it does not want to work the way it is specified in the wiki :smile:. Then I found that the only way to specify a min value was by passing an option object: `{min:5,max:10}`
I think it is quite a handy feature. That's way I decided to implement it.

**Comment by Marak** [comment]
I think there is a bit of room to improve both of these APIs.

I've noticed we are mutating the incoming `options` in a few places we probably should not.

I'll review this PR a bit more and adjust the existing APIs to make everything work a bit better.

Thank you @zpiman

**Comment by zpiman** [comment]
np :smile:

**Comment by zpiman** [comment]
If you don't mind I can have a look at that @Marak

**Comment by Marak** [comment]
You can look at whatever you please my friend! It's open-source.

**Comment by zpiman** [comment]
Ok let me know what you think

**Comment by zpiman** [comment]
@Marak, have a look at it....

**Comment by Marak** [comment]
Marked for review.

Will have this merged soon.

Thank you for your continued patience @zpiman

File: lib/random.js [hunk 1]
```diff
@@ -10,40 +10,46 @@ function Random (faker, seed) {
...
+        defOptions.max = max;
+        defOptions.min = min_options;
+      } else {
+        // User passed only one number... -> max value
+        defOptions.max = min_options;
+      }
+    } else if (typeof min_options === "object") {
+      // User passed an object -> options to be merged into the defaults
+      for (var i in min_options) {
+        if(defOptions.hasOwnProperty(i) && min_options.hasOwnProperty(i)) {
```
Comment (Marak) [inline lib/random.js]: Why use a `hasOwnProperty` call here instead of referencing the object directly?

**Comment by Marak** [comment]
@zpiman - Can you re-open this commit? It's no showing up as `unknown repository` in Github.

**Comment by zpiman** [comment]
Hmm I might have deleted my fork when I was doing a GitHub clean a while ago... I sort of forgot about this pull request let me see if I still have a copy somewhere...

**Comment by zpiman** [comment]
Ok so I might have deleted the branch on all my local repositories. However this feature would need a better re-implementation anyway... I will look into this tomorrow. I will create a new PR.

**Comment by grancalavera** [comment]
Hi @Marak, I just noticed this `faker.random.number` still mutates its argument. I can fix it if it's still  up for grabs...

**Pull request closed without merging**